operators/src/processing/interpolation/mod.rs

```diff
@@ -159,17 +185,16 @@ where
         query: RasterQueryRectangle,
         ctx: &'a dyn QueryContext,
     ) -> Result<BoxStream<'a, Result<Self::Output>>> {
-        ensure!(
-            query.spatial_resolution.x < self.params.input_resolution.x
-                && query.spatial_resolution.y < self.params.input_resolution.y,
-            error::QueryResolutionMustBeSmallerThanInputResolution {
-                query_resolution: query.spatial_resolution,
-                input_resolution: self.params.input_resolution,
-            }
-        );
+        // do not interpolate if the source resolution is already fine enough
+        if query.spatial_resolution.x >= self.input_resolution.x
+            && query.spatial_resolution.y >= self.input_resolution.y
+        {
+            // TODO: should we use the query or the input resolution here?
+            return self.source.query(query, ctx).await;
+        }
```

Comment:
I guess query as you did, otherwise we cannot load less data depending on the query input.
This would mean, we always query with the lowest pyramid level right?

Reply:
no, it means we query with the pyramid level that has the next finer resolution as the query. Could also be the highest pyramid level or some level in the middle.

Reply:
Why the next finer? The input resolution is fixed, right?

Reply:
If we would use it then yes. I don't know why it would correspond to pyramid levels, though. The user could specify any inputResolution. It's independent of any pyramids.